cli/src/opts.rs

```diff
@@ -51,6 +51,7 @@ impl Opts {
     /// The address that the service should be bound to.
     pub fn addr(&self) -> SocketAddr {
         self.socket_addr
+            .unwrap_or_else(|| "127.0.0.1:7878".parse().unwrap())
```

Comment:
nit: Would you mind constructing the `SocketAddr` via [`new`](https://doc.rust-lang.org/std/net/struct.SocketAddrV4.html#method.new), rather than a `.parse().unwrap()` chain?

Reply:
I originally had that but decided to use the parse method to keep the textual representation so it's easier to search for people unfamiliar with rust but I can revert back to Sockaddr::new()..

Reply:
Updated.